samples/modelbuilder/ObjectDetection_StopSigns/StopSignDetectionML.Model/ConsumeModel.cs

```diff
@@ -0,0 +1,63 @@
+// This file was auto-generated by ML.NET Model Builder. 
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.IO;
+using Microsoft.ML;
+using StopSignDetectionML.Model;
+using System.Net.Http;
+using Microsoft.ML.Trainers;
+using System.Web;
+using System.Net;
+
+namespace StopSignDetectionML.Model
+{
+    public class ConsumeModel
+    {
+        private static Lazy<PredictionEngine<ModelInput, ModelOutput>> PredictionEngine = new Lazy<PredictionEngine<ModelInput, ModelOutput>>(CreatePredictionEngine);
+
+        private static WebClient client = new WebClient();
+
+        public static string MLNetModelPath = Path.GetFullPath(@"../../../../StopSignDetectionML.Model/MLModel.zip");
+
+        // For more info on consuming ML.NET models, visit https://aka.ms/mlnet-consume
+        // Method for consuming model in your app
+        public static ModelOutput Predict(ModelInput input)
+        {
+            ModelOutput result = PredictionEngine.Value.Predict(input);
+            return result;
+        }
+
+        public static PredictionEngine<ModelInput, ModelOutput> CreatePredictionEngine()
```

Comment:
What's our recommended best practice around `PredictionEngine.Dispose: one static and reuse it or tie it to some other instance and dispose?  It's a heavy object so I can see why we wouldn't want to create one each time we use it, but I get anxious when I see Disposable objects being left for finalization to cleanup.

Reply:
The prediction engine isn't thread safe. So both things you suggested have been done. If we can reuse it we should, but if its in a threaded environment we can't.

Reply:
> both things you suggested have been done

I'm not asking about PredictionEngine itself, but how it is used.

If it's not thread safe then our sample code shouldn't be wrapping it in a static.  Maybe since there's not much happening in this class any more we could move CreatePredictionEngine into Program.cs, then just do this:
```c#
using (var predictionEngine = CreatePredictionEngine())
{
    var predictionResult = ConsumeModel.Predict(sampleData);
}
```

Would that still achieve the goal of the sample?

Alternatively, if we're trying demonstrate thread-safe prediction we could keep the static class and use a ThreadLocal for the engine.

Reply:
Not sure about this, but this is the code that's generated by Model Builder

Reply:
This is the code generated by Model Builder, so if there's issues with PredictionEngine then might have to work with tooling team on changing code generation as well

Reply:
If its generated by model builder does it need to be checked in? Or did you just want an example of what it will look like?

Reply:
I still need the sample to be able to run (users can download the whole sample or follow the tutorial step by step)

Reply:
So I think that looking into changing the code generated by the model builder would be good since @ericstj is right that we shouldn't have non-thread safe code be static, but since the code is auto generated and not anything you have modified I don't think we should block that here.

@ericstj thoughts?

Reply:
I hadn't realized this was auto-generated.  If that's the case, let's revert the code to the exact version that's auto-generated and treat the characteristics of the auto-generated code as a separate issue.  cc @eerhardt @JakeRadMSFT